dao/progress.go

```diff
@@ -30,12 +30,40 @@ func (d progressDao) GetProgressesForUser(userID uint) (r []model.StreamProgress
 	return r, DB.Where("user_id = ?", userID).Find(&r).Error
 }
 
+func filterProgress(progresses []model.StreamProgress, watched bool) []model.StreamProgress {
+	var result []model.StreamProgress
+	for _, progress := range progresses {
+		if progress.Watched == watched {
+			result = append(result, progress)
+		}
+	}
+	return result
+}
+
 // SaveProgresses saves a slice of stream progresses. If a progress already exists, it will be updated.
 func (d progressDao) SaveProgresses(progresses []model.StreamProgress) error {
-	return DB.Clauses(clause.OnConflict{
-		Columns:   []clause.Column{{Name: "stream_id"}, {Name: "user_id"}}, // key column
-		DoUpdates: clause.AssignmentColumns([]string{"progress"}),          // column needed to be updated
-	}).Create(progresses).Error
+	noWatched := filterProgress(progresses, false)
```

Comment:
Please help me understand, why is this change required? Can't we just add `watched` to the `assignmentColumns`?

Reply:
If I just add it to the columns it will overwrite the existing value. If the user already watched the whole VoD, but then skips to the middle, to rewatch something, the whole VoD will be marked as unwatched

Reply:
makes sense, thanks! can you add that as a comment as well?